src/rulesets/LinearAlgebra/factorization.jl

```diff
@@ -70,20 +70,69 @@ end
 ##### `cholesky`
 #####
 
-function rrule(::typeof(cholesky), X::AbstractMatrix{<:Real})
-    F = cholesky(X)
-    function cholesky_pullback(Ȳ::Composite)
-        ∂X = if F.uplo === 'U'
-            chol_blocked_rev(Ȳ.U, F.U, 25, true)
-        else
-            chol_blocked_rev(Ȳ.L, F.L, 25, false)
-        end
-        return (NO_FIELDS, ∂X)
+function rrule(::typeof(cholesky), A::Real, uplo::Symbol=:U)
+    C = cholesky(A, uplo)
+    function cholesky_Real_pullback(Δ::Composite)
+        return NO_FIELDS, Δ.factors[1, 1] / (2 * C.U[1, 1]), DoesNotExist()
     end
-    return F, cholesky_pullback
+    return C, cholesky_Real_pullback
+end
+
+function rrule(
+    ::typeof(cholesky), A::Diagonal{<:Real}, ::Val{false}=Val(false); check::Bool=true,
+)
+    C = cholesky(A, Val(false); check=check)
+    function cholesky_Diagonal_pullback(Δ::Composite)
+        issuccess(C) || throw(PosDefException(C.info))
+        Ā = Diagonal(diag(Δ.factors) .* inv.(2 .* C.factors.diag))
+        return NO_FIELDS, Ā, DoesNotExist()
+    end
+    return C, cholesky_Diagonal_pullback
+end
+
+# The appropriate cotangent is different depending upon whether A is Symmetric / Hermitian,
+# or just a StridedMatrix.
+# Implementation due to Seeger, Matthias, et al. "Auto-differentiating linear algebra."
+function rrule(
+    ::typeof(cholesky),
+    A::Union{Symmetric{<:Real, <:StridedMatrix}, Hermitian{<:Real, <:StridedMatrix}},
+    ::Val{false}=Val(false);
+    check::Bool=true,
+)
+    C = cholesky(A, Val(false); check=check)
+    function cholesky_SymHerm_pullback(Δ::Composite)
+        Ā, U = _cholesky_pullback_shared_code(C, Δ)
+        Ā = BLAS.trsm!('R', 'U', 'T', 'N', one(eltype(Ā)) / 2, U.data, Ā)
+        return NO_FIELDS, _symhermtype(A)(Ā), DoesNotExist()
+    end
+    return C, cholesky_SymHerm_pullback
+end
+
+function rrule(
+    ::typeof(cholesky), A::StridedMatrix{<:Real}, ::Val{false}=Val(false); check::Bool=true,
+)
+    C = cholesky(A, Val(false); check=check)
+    function cholesky_StridedMatrix_pullback(Δ::Composite)
+        Ā, U = _cholesky_pullback_shared_code(C, Δ)
+        Ā = BLAS.trsm!('R', 'U', 'T', 'N', one(eltype(Ā)), U.data, Ā)
+        Ā[diagind(Ā)] ./= 2
+        return (NO_FIELDS, UpperTriangular(Ā), DoesNotExist())
+    end
+    return C, cholesky_StridedMatrix_pullback
+end
+
+function _cholesky_pullback_shared_code(C, Δ)
+    issuccess(C) || throw(PosDefException(C.info))
+    U = C.U
+    Ū = Δ.U
```

Comment:
This should be handled by `getproperty` I believe -- it flips stuff appropriately for the `uplo` property of the `cholesky`. Since the user-facing interface no longer provides an option to set `uplo`, I think we might be fine. Does this seem reasonable to you?